### PR TITLE
[ONNX] Support .item() export & NumberType to tensor conversion

### PR DESCRIPTION
**Pull request by BowenBao**

### CODE REVIEW

File: aten/src/ATen/core/jit_type.h [hunk 1]
```diff
@@ -1555,6 +1555,8 @@ inline TypePtr TensorType::fromNumberType(TypePtr typ) {
     return TensorType::createContiguous(at::kDouble, at::kCPU, {});
   } else if (typ->isSubtypeOf(BoolType::get())) {
     return TensorType::createContiguous(at::kBool, at::kCPU, {});
+  } else if (typ->kind() == NumberType::Kind) {
+    return TensorType::create(c10::nullopt, at::kCPU, {}, c10::nullopt);
```
Comment (shubhambhokare1) [inline aten/src/ATen/core/jit_type.h:1559]: Is there a way to add a test to demonstrate this behavior?

**Review by shubhambhokare1 — COMMENTED** [review]
item op looks good to me, I'm not exactly sure what the NumberType related change is (maybe description would be helpful)

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 2]
```diff
@@ -4026,6 +4026,19 @@ def forward(self, x, y, z, ind):
         ind = torch.tensor(-2, dtype=torch.long)
         self.run_test(GetItemModel(), (x, y, z, ind))
 
+    def test_item(self):
+        class M(torch.nn.Module):
+            def comp(self, a: int, b: int):
```
Comment (neginraoof) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:4031]: What is the use of 'comp' ?

Comment (BowenBao) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:4031]: good catch, test code while debugging, removed.

File: aten/src/ATen/core/jit_type.h (hunk 1 above)
Comment (BowenBao) [reply aten/src/ATen/core/jit_type.h:1559]: this is created in the IR graph with `tensor.item()`

**Review by shubhambhokare1 — APPROVED** [review]
LGTM

**Pull request merged**